### DISCUSSION TITLE
Is providing support for dev containers on the roadmap?

### DISCUSSION THREAD

**Discussion by aalexandersson** [posit-dev/positron#3747] [Feedback]
Is providing support for [dev containers](https://containers.dev/) on the roadmap? The [Positron wiki](https://github.com/posit-dev/positron/wiki) states that Positron might not be a good fit for you today if...

> you use remote development features (e.g. dev containers, remote SSH); these are not supported in Positron yet. However, providing tools for Remote SSH is on our roadmap.

**Comment by jmcphers** [comment]
Yes, we're interested in doing this. However, it's not as simple as picking up the feature from Code OSS; the remote development features of Visual Studio Code are closed source Microsoft software not licensed for use outside the commercially licensed distribution. This includes the vscode-server component that gets installed inside the container to supply the remote end of the connection. You can read more about this here:

https://code.visualstudio.com/docs/remote/faq#_license-and-privacy

This is the same reason that other Code OSS forks like VSCodium don't have dev containers; see here for a discussion about that: https://github.com/VSCodium/vscodium/discussions/1471

So, doable but not on the near term roadmap since it'll likely be a larger investment.

**Comment by jmcphers** [comment]
For other folks landing here from search, there's an issue tracking this now: https://github.com/posit-dev/positron/issues/4691
